engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbUserTaskState.java

```diff
@@ -110,18 +107,4 @@ public UserTaskRecord getUserTask(final long key) {
     final UserTaskRecordValue userTask = userTasksColumnFamily.get(userTaskKey);
     return userTask == null ? null : userTask.getRecord();
   }
-
-  @Override
-  public UserTaskRecord getUserTask(final long key, final Map<String, Object> authorizations) {
-    final UserTaskRecord userTask = getUserTask(key);
-    if (userTask != null
-        && getAuthorizedTenantIds(authorizations).contains(userTask.getTenantId())) {
-      return userTask;
-    }
-    return null;
-  }
-
-  private List<String> getAuthorizedTenantIds(final Map<String, Object> authorizations) {
-    return (List<String>) authorizations.get(Authorization.AUTHORIZED_TENANTS);
```

Comment:
:x: We should leave the tenant authorization check in the UserTaskState. This was a design decision the ZPA team made while developing the multi-tenancy epic.

Why we decided this: Multi-tenancy is a data handling/isolation concept. Since the `*State` classes are responsible for handling data in Zeebe, it made the most sense to place the tenant authorization check. This way, the processors never get access to data a command is not authorized to handle, i.e. data never exits the state if the requester is not authorized to receive it.

If we decide to place the tenant authorization check in the processors, it becomes more difficult to maintain and correctly implement this for each processor. 

We actually have a tech debt issue to improve how we handle the tenant data and how it reaches the state here: https://github.com/camunda/zeebe/issues/14256.

Reply:
Btw, I came across this because we implemented it this way for multiple processors here https://github.com/camunda/zeebe/pull/14553. The PR already states that it would be better to have this in the state handling itself but it still led me to do this here as well 😅

Reply:
Reverted this and force-pushed.

Reply:
Yeah, that was a quick fix since we don't keep tenant information in the `ElementInstanceState` directly. That makes #14256 even more important.